plugins/modules/ipa_host.py

```diff
@@ -236,16 +238,17 @@ def ensure(module, client):
 
     ipa_host = client.host_find(name=name)
     module_host = get_host_dict(description=module.params['description'],
-                                force=module.params['force'], ip_address=module.params['ip_address'],
+                                force=module.params['force'],
+                                ip_address=module.params['ip_address'],
                                 ns_host_location=module.params['ns_host_location'],
                                 ns_hardware_platform=module.params['ns_hardware_platform'],
                                 ns_os_version=module.params['ns_os_version'],
                                 user_certificate=module.params['user_certificate'],
                                 mac_address=module.params['mac_address'],
-                                random_password=module.params.get('random_password'),
+                                random_password=module.params['random_password'],
                                 )
     changed = False
-    if state in ['present', 'enabled', 'disabled']:
+    if state in ['present', 'enabled']:
```

Comment:
This change looks wrong. `disabled` should (apparently) be a special case of `present`, so the host exists, but is disabled.

Reply:
yeah, but if there are no host found It would create one which seems to be weird because `enabled` and `disabled` are states for managing already existing hosts so I created an additional if statement for that case

Reply:
I don't think it's weird. But I think there should also be another state, like `disabled-or-absent`. Depending on what you want to achieve `disabled` or `disabled-or-absent` is the right state to use.

Reply:
I added `force_creation` parameter for that case